File: sdk/ocaml/lib/cloud_gpu.ml
(** Cloud GPU: GPU instance management *)

(** PUT /api/service/{id}/vms/reboot - Reboot virtual server *)
let reboot client params =
  HttpClient.put client "/api/service/{id}/vms/reboot" params

(** PUT /api/service/{id}/vms/stop - Stop virtual server *)
let stop client params =
  HttpClient.put client "/api/service/{id}/vms/stop" params

(** PUT /api/service/{id}/vms/start - Start virtual server *)
let start client params =
  HttpClient.put client "/api/service/{id}/vms/start" params

(** GET /api/service/{id}/vms/firewall - Get firewall rules *)
let get_firewall_rules client params =
  HttpClient.get client "/api/service/{id}/vms/firewall" params

(** POST /api/service/{id}/vms/firewall - Add firewall rules *)
let add_firewall_rules client params =
  HttpClient.post client "/api/service/{id}/vms/firewall" params

(** DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule *)
let remove_firewall_rule client params =
  HttpClient.delete client "/api/service/{id}/vms/firewall/{position}" params

File: sdk/dart/lib/src/cloud_gpu.dart
import 'http_client.dart';

/// Cloud GPU: GPU instance management
class CloudGpuResource {
  final HttpClient httpClient;
  CloudGpuResource(this.httpClient);

  /// PUT /api/service/{id}/vms/reboot - Reboot virtual server
  Future<Map<String, dynamic>> reboot([Map<String, dynamic>? params]) async {
    return httpClient.put('/api/service/{id}/vms/reboot', params);
  }

  /// PUT /api/service/{id}/vms/stop - Stop virtual server
  Future<Map<String, dynamic>> stop([Map<String, dynamic>? params]) async {
    return httpClient.put('/api/service/{id}/vms/stop', params);
  }

  /// PUT /api/service/{id}/vms/start - Start virtual server
  Future<Map<String, dynamic>> start([Map<String, dynamic>? params]) async {
    return httpClient.put('/api/service/{id}/vms/start', params);
  }

  /// GET /api/service/{id}/vms/firewall - Get firewall rules
  Future<Map<String, dynamic>> getFirewallRules([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/service/{id}/vms/firewall', params);
  }

  /// POST /api/service/{id}/vms/firewall - Add firewall rules
  Future<Map<String, dynamic>> addFirewallRules([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/service/{id}/vms/firewall', params);
  }

  /// DELETE /api/service/{id}/vms/firewall/{position} - Remove firewall rule
  Future<Map<String, dynamic>> removeFirewallRule([Map<String, dynamic>? params]) async {
    return httpClient.delete('/api/service/{id}/vms/firewall/{position}', params);
  }

}
